http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala

```diff
@@ -0,0 +1,150 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package db
+
+import java.time.Instant
+
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.google2.{DiskName, ZoneName}
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.api._
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.mappedColumnImplicits._
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.{dummyDate, unmarshalDestroyedDate}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+import scala.concurrent.ExecutionContext
+
+final case class PersistentDiskRecord(id: Long,
+                                      googleProject: GoogleProject,
+                                      zone: ZoneName,
+                                      name: DiskName,
+                                      googleId: Option[GoogleId],
+                                      samResourceId: DiskSamResourceId,
+                                      status: DiskStatus,
+                                      creator: WorkbenchEmail,
+                                      createdDate: Instant,
+                                      destroyedDate: Instant,
+                                      dateAccessed: Instant,
+                                      size: DiskSize,
+                                      diskType: DiskType,
+                                      blockSize: BlockSize)
+
+class PersistentDiskTable(tag: Tag) extends Table[PersistentDiskRecord](tag, "PERSISTENT_DISK") {
```

Comment:
I think you need to define an `index` to correspond to `IDX_PERSISTENT_DISK_UNIQUE`

Reply:
We typically haven't represented all our indexes in the slick mapping code. The indexes are still present in the database. IIRC they are only useful in slick when generating the schema from slick code, which we don't do